src/tool/hpcrun/control-knob.c

```diff
@@ -39,10 +39,13 @@ control_knob_register(char *name, char *value, control_knob_type type)
     iter = (control_knob_t*) malloc(sizeof(control_knob_t));
     iter->name = strdup(name);
     iter->type = type;
+    iter->value = strdup(value);
```

Comment:
It can be simplified. Both branches execute `iter->value = strdup(value);`

Reply:
I was planning to do that, but my bug arose because of me trying to pull things out of if().
This way I think it is easier to read, and all functionality is self-contained in if/else.

I can change it if you want.

Reply:
please put the assignment iter->value=strdup(value) after the if. eliminate the assignment in the if branch and remove the else branch entirely.